src/dask_awkward/lib/str.py

```diff
@@ -597,10 +603,10 @@ def replace_substring(
 @always_highlevel
 def replace_substring_regex(
     array: Array,
-    pattern,
-    replacement,
+    pattern: str,
+    replacement: str | bytes,
     *,
-    max_replacements=None,
+    max_replacements: int | None = None,
```

Comment:
some formatters prefer Optional[int]. I don't know if there's any consequence of choosing one over the other.

Reply:
with `from __future__ import annotations` the current ["blessed" syntax](https://stackoverflow.com/questions/69440494/python-3-10-optionaltype-or-type-none) is `foo | None` over `Optional[foo]`. Some interesting arguments in that SO thread, of course!